src/settings/components/TemplateEditor/TemplateEditor.js

```diff
@@ -128,6 +134,32 @@ class TemplateEditor extends React.Component {
     this.setState({ cursorPosition: cursorPosition + tags.length });
   };
 
+  insertRepeatingTokens = (tokens, tag) => {
+    let tags = '';
+    const startTag = `{{#${tag}}}`;
+    const endTag = `{{/${tag}}}`;
+
+    const editor = this.quill.current.getEditor();
+    const editorContent = editor.getText();
+    const loopExists = includes(editorContent, startTag) && includes(editorContent, endTag);
+
+    if (loopExists) {
```

Comment:
I would suggest simplifying things there a little bit

```
let tags = map(tokens, (token) => token); or just Object.values(tokens)

if (!loopExists) {
  tags = [startTag, ...tokens, endTag];
}

tags = tags.join('\n');
```
Does it make sense?

Reply:
No, take a look on if statement. Depending on condition we should add token with '\n' on different positions. Also in case if loop exists we should add end tag with '\n'. If i will follow your approach the code will look like:
```
tags = loopExists 
  ? tags = map(tokens, token => `{{${token}}}\n`)
  : [startTag, map(tokens, token => `\n{{${token}}}`), '\n', endTag];

editor.insertText(cursorPosition, tags.join(''));
```
Not sure that this code is more readable.
Moreover in this case we receive strange behavior for the cursor position.

Reply:
Agree.